offlineimap/imapserver.py

```diff
@@ -198,7 +198,7 @@ def __md5handler(self, response):
         challenge = response.strip()
         self.ui.debug('imap', '__md5handler: got challenge %s'% challenge)
 
-        passwd = self.__getpassword()
+        passwd = self.__getpassword().encode('ascii')
```

Comment:
* I find `bytearray` less clearer than `encode()` (which is called by bytearray, anyway).
* Above code **is** fixing the type issue in v3.4 since `hmac.new()` is passed a byte type.
* We are in the cycle of a new major release. Because of this, we should make breaking changes now. We want to support both py3 and py2. My biggest worry is that it will be confusing and unecessary made complex to support both bytes and str. We can decide to enforce users to use unicode in their configuration file so that we make our job much easier in the code.

I'll write patches so we can discuss on them.